### DISCUSSION TITLE
Has anyone used this with c#?

### DISCUSSION THREAD

**Discussion by scottc-WellSky** [microsoft/regorus#282] [Q&A] [answered]
I have an issue and I assume it's because of the c# interop, but I'm not positive. I can add the policy and set the input, but when calling EvalQuery I always get back an empty result:

```
var regoEngine = new RegorusPolicyEngine();
regoEngine.AddPolicyFromPath(HostingEnvironment.MapPath("~/App_Data/Rego/Policies/"));
regoEngine.AddDataJson("{}");

regoEngine.SetInputJson("{\"assessment\": {\"id\": \"assessment123\",\"requires_consent\": true,\"created_by\": \"user123\"},\"consent_forms\": [{\"consent_id\": \"consent123\",\"state\": \"Approved\",\"effective_date\": \"2023-01-01T00:00:00Z\",\"expiration_date\": \"2024-12-31T23:59:59Z\",\"is_revoked\": false}]}");

var results = regoEngine.EvalQuery("data.assessment_consent.has_consent");
````

results = {}

**Comment by scottc-WellSky** [comment, marked as answer]
Never mind. I had a naming faux paux. assessment_consent != assessment.consent. All working well now.

**Comment by anakrish** [comment]
@scottc-WellSky We are using it internally at MSFT from C#. Do let us know if you spot anything amiss or have suggestions for improvement. cc @yangjie-msft
